#!/usr/bin/env python3
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator

PACKAGE = "sr_piezo_feedback"

gen = ParameterGenerator()

gen.add("contact_time", double_t, 0, "Fading time after contact",  0.25, 0.25,  2.0)

gen.add("max_amplitude", double_t, 0, "Maximum amplitude on output signal", 1.0, 0.5, 1.0)
gen.add("min_amplitude", double_t, 0, "Minimum amplitude on output signal", 0.2, 0.0, 0.4)
gen.add("max_frequency", double_t, 0, "Maximum frequency on output signal", 80.0, 30.0, 150.0)
gen.add("min_frequency", double_t, 0, "Minimum frequency on output signal", 1.0, 0.1, 10.0)

gen.add("pdc_threshold", double_t, 0, "pdc_threshold", 1.0, 5.0, 10.0)
gen.add("pdc_saturation", double_t, 0, "pdc_saturation", 50.0, 100.0, 200.0)
gen.add("pdc_mapping_exponent", double_t, 0, "pdc_mapping_exponent", 0.1, 2.0, 10.0)
gen.add("pdc_output_weight", double_t, 0, "pdc_output_weight", 0.1, 2.0, 5.0)

gen.add("pac_threshold", double_t, 0, "pac_threshold", 5.0, 10.0, 20.0)
gen.add("pac_saturation", double_t, 0, "pac_saturation", 1000.0, 2000.0, 3000.0)
gen.add("pac_mapping_exponent", double_t, 0, "pac_mapping_exponent", 0.1, 5.0, 10.0)
gen.add("pac_output_weight", double_t, 0, "pac_output_weight", 0.1, 0.3, 5.0)

exit(gen.generate(PACKAGE, "sr_piezo_feedback_node", "SrPiezoFeedbackBiotac"))
